### ISSUE TITLE
feature: Updating Dates in the datatable

### ISSUE THREAD

**Issue by gryBox** [bokeh/bokeh#5402]
Hi - Editing datatables with dates

Before editing the date
![image](https://cloud.githubusercontent.com/assets/16763201/19662005/937b1266-99fb-11e6-9125-3da6c49b7e09.png)

during
![image](https://cloud.githubusercontent.com/assets/16763201/19662021/9f75ad1a-99fb-11e6-8588-b30fdc1dae7d.png)

Not exactly the what an end user wants to see when they change dates :)

minimally: I think the format should follow some human readable from the `DateFormatter` 

ideally: A datepicker would pop up similar to the pic below (from [handsontable](https://handsontable.com/examples.html))
![image](https://cloud.githubusercontent.com/assets/16763201/19662094/dac8377a-99fb-11e6-94c0-12d40837bc4b.png)

The code:
```
from datetime import date
from random import randint

from bokeh.io import output_file, show
from bokeh.layouts import widgetbox
from bokeh.models import ColumnDataSource
from bokeh.models.widgets import DataTable, DateFormatter, TableColumn

output_file("data_table.html")

data = dict(
        dates=[date(2014, 3, i+1) for i in range(10)],
        downloads=[randint(0, 100) for i in range(10)],
    )
source = ColumnDataSource(data)

columns = [
        TableColumn(field="dates", title="Date", formatter=DateFormatter(format='yy-mm-dd')),
        TableColumn(field="downloads", title="Downloads"),
    ]
data_table = DataTable(source=source, columns=columns, editable=True, width=400, height=280)

show(widgetbox(data_table))
```

**Comment by birdsarah** [comment]
Thanks for the report @gryBox. 

Here's the PR that added editable cells: https://github.com/bokeh/bokeh/pull/5203

Here's an example of slickgrid with datepicker http://6pac.github.io/SlickGrid/examples/example-select2-editor.html so I'm hopeful this is not too bad to add.

**Comment by birdsarah** [comment]
Correct: #5203 made sure edits to cells updated the data source I guess.

**Comment by pmreyes2** [comment]
Since the date data source in ColumnDataSource is epoch milliseconds, one can also envision a datetimepicker like the one from jquery-datetimepicker, where time elements could be optional.
![image](https://cloud.githubusercontent.com/assets/3052100/21163888/3d3084ae-c15b-11e6-8932-028175a66a3f.png)

**Comment by gryBox** [comment]
That would be cool.

Itay Livni
